src/Tools/dotnet-monitor/Egress/FileSystem/FileSystemEgressExtension.cs

```diff
@@ -38,6 +38,16 @@ public FileSystemEgressExtension(IEgressConfigurationProvider configurationProvi
             EgressArtifactSettings settings,
             Func<Stream, CancellationToken, Task> action,
             CancellationToken token)
+        {
+            return await EgressArtifact(providerName, settings, action, false, token);
+        }
+
+        public async Task<EgressArtifactResult> EgressArtifact(
```

Comment:
Undo everything related to this "new" method since validation is handled separately.

Reply:
Try to remove the validation code from here and create a separate method for validation that is sharable among `EgressArtifact` and `ValidateProviderAsync`. It can even use https://github.com/dotnet/dotnet-monitor/blob/main/src/Tools/dotnet-monitor/DataAnnotationValidateOptions.cs if you want to deduplicate code.